### DISCUSSION TITLE
Option to adjust cursor bar's width

### DISCUSSION THREAD

**Discussion by wzxu** [ghostty-org/ghostty#2657] [Q&A] [answered]
As it is right now, the cursor with `cursor-style = bar` is fixed to be a 1px line even on Retina (@2x) displays, and it's just way too thin to be usable for me.
<img width="172" alt="22319" src="https://github.com/user-attachments/assets/6b70fedb-9be6-415d-ae1b-22a3e2aaf854">

Alacritty supports setting `[cursor] thickness` so I can make it even slightly wider than just a typical 2px line.
Bonus point if we can also round the rectangle a little bit to make it visually pleasing like Warp terminal's cursor.
<img width="140" alt="34395" src="https://github.com/user-attachments/assets/1a804f44-803c-4df4-9b4c-8397d705dc69">

**Comment by mitchellh** [comment]
I like the suggestion.

> Bonus point if we can also round the rectangle a little bit to make it visually pleasing like Warp terminal's cursor.

I like this as well. 

Will wait a bit to hear any feedback but planning to convert to an issue.

**Comment by luisdavim** [comment]
The thin cursor is especially problematic in vim, at least with my theme, when the cursor is next to the gutter or a vertical split separator.

**Comment by piechologist** [comment, marked as answer]
We already have the setting `adjust-cursor-thickness = 50%` which does the trick for me. The cursor isn't rounded but you can make it *really* thick.

**Reply by wzxu** [reply]
Huh I must've glossed over that in the default config list. 
This option alone should make it okay for me. The rounding is just a nice-to-have.

**Reply by wzxu** [reply]
Thanks for the hint!
